OneCommunityGlobal/HighestGoodNetworkApp
# show badges in weekly report

# Description
Jae: Make most difficult to earn badges show up by a person’s name on Reports Page
There are certain badges that are very hard to earn. When a person earns these, I’d like them to post to the right of the person’s other details on the Reports → Weekly Summaries Reports page. This is so that Management can see these badges were earned by that person and acknowledge them for it
- Make visible to anyone who has access to the page
- Make it remain on whatever weekly tab it is awarded on, just like a weekly summary stays on that tab and can be referenced up to two weeks past.
- Make it so multiple badges can be shown on this page if they happen to be earned in the same week
- Integrate this into the Other Links → Badge Management → Badge Development page. Add a column for “Reports Page Notification”. Checking a box here makes that badge one that shows up on the Weekly Summaries Reports page tab for that week when earned

## Related PRS (if any):
This frontend PR is related to the [#470 backend PR](https://github.com/OneCommunityGlobal/HGNRest/pull/470).

## Main changes explained:
- The weekly summary report page will show the badges earned within the week
- Add an attribute 'showReport' to the badges, If its value is false (the report notification of a badge is not checked), it will not show in the report.

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally (after update the backend PR)
3. log as owner/admin user
4. go to the profile page of a user, assign a badge A.
5. Other Links → Badge Management → Badge Development: click the checkbox to make the badge A appears on the report page.
6. Reports → Weekly Summaries Reports --> This Week, search the user and see the badge A appears under the username.

## Screenshots or videos of changes:
<img width="1502" alt="1" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/9314962/a8f32ecf-1446-4952-b0ef-5e7419a1a877">
<img width="529" alt="2" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/9314962/8bd6dd47-1c35-4a78-b0d9-992612cd02e0">
